modules/shared/k8shelpers/kube-config-watcher.go

```diff
@@ -138,7 +116,7 @@ func (w *KubeConfigWatcher) start() {
 			case fsEv.Op&fsnotify.Create == fsnotify.Create:
 				// Load config
 				w.mu.Lock()
-				kubeConfig, err := clientcmd.LoadFromFile(w.kubeconfigPath)
+				kubeConfig, err := clientcmd.LoadFromFile(fsEv.Name)
```

Comment:
This might cause issues as it will overwrite the global merged `*api.Config` object by a `*api.Config` with the configuration from the changed file ... need to fix this.

Reply:
I think the simplest way to modify the watcher is to have it handle the config "merging" internally so that downstream consumers continue to see just one config object. If there were unit tests that would be an easier (sorry!). Do you want to try to do it as part of this PR or hand off that piece?